### PR TITLE
Basic layout for Flask dev/test

### PR DESCRIPTION
**Pull request by jayakasadev**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
 
 LABEL maintainer="bharadwajr07"
+WORKDIR /app
+
+# avoid stuck build due to user prompt
+ARG DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update -y && \
-    apt-get install -y python-pip python-dev
+    apt-get install -y python3.9 python3.9-dev python3-pip python3-wheel build-essential
```
Comment (jayakasadev) [inline Dockerfile:12]: may be able to prune this down to the essentials

File: src/main/app.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+app = Flask(__name__)
+app.logger.setLevel(logging.INFO)
+NAME = "Bharadwaj"
+
+name = None
+
+
+@app.route('/', methods=['GET'])
+async def index():
```
Comment (jayakasadev) [inline src/main/app.py:12]: if you don't know how to use async, delete it from here and remove the `[async]` from `requirements.txt`

File: src/main/app.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    app.logger.info("Received GET / request")
+    # \n to avoid % escape
+    return 'Hello {}\n'.format(NAME), 200
+
+
+@app.route('/set_name', methods=['POST'])
+async def set_name():
```
Comment (jayakasadev) [inline src/main/app.py:19]: same as above

File: src/test/test_app.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+def test_set_name(client):
+    expected = {
+        "message": "Name changed!",
+        "old_name": None,
+        "new_name": "TEST",
+    }
+    response = client.post(
```
Comment (jayakasadev) [inline src/test/test_app.py:7]: use the client fixture to make get/post calls

File: Dockerfile (hunk 1 above)
Comment (bharadwajr567) [reply Dockerfile:12]: Got it.

**Pull request merged**